### DISCUSSION TITLE
Size of `Operator` in binary

### DISCUSSION THREAD

**Discussion by thesuhas** [bytecodealliance/wasm-tools#1801] [Q&A] [answered]
Is there anyway one can determine the size of a `wasmparser::Operator` in binary on encoding? I would like this information to find the offset of the instruction in the binary but I don't see how one could obtain it.
An idea is to create an empty `wasm_encoder::Module` and create an empty `CodeSection`. Encode the respective instruction in it and use the size of the code section before and after encoding the instruction to find the size in binary

**Comment by fitzgen** [comment]
I don't believe we expose an easier way of doing it now than what you've sketched but I don't think there is any fundamental reason we couldn't do better

**Comment by fitzgen** [comment, marked as answer]
Actually, we implement `wasm_encoder::Encode` for `wasm_encoder::Instruction` so I think it would be slightly easier (compared to your OP) to do something like

```rust
use wasm_encoder::{Encode, reencode::Reencode};

let op: wasmparser::Operator = todo!();
let inst: wasm_encoder::Instruction = wasm_encoder::RoundtripReencoder.instruction(op)?;

let mut buf = vec![];
inst.encode(&mut buf);

Ok(buf.len())
```

**Reply by fitzgen** [reply]
I don't think we could really do anything better than this for an arbitrary `wasmparser::Operator`.

If you are parsing the operator yourself, I think you could look at the `OperatorReader`'s underlying `BinaryReader`'s current offset before and after parsing, and then save that info somewhere.
